samples/rpc/src/main.c

```diff
@@ -60,6 +57,13 @@ void main(void)
 	client->on_connect = golioth_on_connect;
 	golioth_system_client_start();
 
+	assert(IS_ENABLED(CONFIG_GOLIOTH_RPC));
```

Comment:
I'm not fan of this `assert()`, as it bloats code just to double check if the `prj.conf` configuration option is applied. If it won't be set, then there will be a warning in cmake configuration step already.